.github/workflows/mfem-analysis.yml

```diff
@@ -0,0 +1,91 @@
+# Copyright (c) 2010-2021, Lawrence Livermore National Security, LLC. Produced
+# at the Lawrence Livermore National Laboratory. All Rights reserved. See files
+# LICENSE and NOTICE for details. LLNL-CODE-806117.
+#
+# This file is part of the MFEM library. For more information and source code
+# availability visit https://mfem.org.
+#
+# MFEM is free software; you can redistribute it and/or modify it under the
+# terms of the BSD-3 license. We welcome feedback and contributions, see file
+# CONTRIBUTING.md for details.
+
+name: build-analysis
```

Comment:
What is build analysis for?

Reply:
There used to be two jobs in this section. This is were I put checks that need a build on MFEM (unlike repo-check).
Ultimately, other checks like memcheck could go in this section.

Reply:
(coverage used to be there, but I moved it to the checks-and-tests (to be renamed) because coverage is accumulated across the regular targets.

Reply:
I'd like to propose a reorganization to keep things simple:

Just two workflows: 

1. `Checks` (scripts that do not require a build of mfem)
2. `Tests` (scripts that first build mfem)

The current 4 items in `repo-check` will be just moved in `Checks` and `copyright-check` can be renamed to just `copyright`.

The 10 items in `build-analysis` and `builds-and-tests` will be moved in `Tests` and we can rename them as follows:

- `ubuntu-latest-debug-mpi(NO)-make` -> `linux-ser-dbg`
- `ubuntu-latest-optim-mpi(YES)-make` -> `linux-par-opt`
- `macos-latest-debug-mpi(YES)-make` -> `mac-par-dbg`
- `ubuntu-latest-optim-mpi(YES)-cmake` -> `cmake`

What do you think?

Reply:
I am also wondering if this is the right organization, or we should have just one but more complex workflow?

For example in such workflow the `Checks` can be run first, and only if they pass (except `branch-history`) we run the 9 `Tests` with linux, mac and cmake, and after they pass, we run `gitignore`)?

Is that the idea of workflows, or should we keep it simple?

Am I correct that if `code-style` fails now the rest of the tests will run?